romanisim/nonlinearity.py

```diff
@@ -199,35 +148,19 @@ class NL:
     tightly coupled to the apportionment mechanism and so I haven't
     explored that adequately.
     """
-    def __init__(self, coeffs):
+    def __init__(self, coeffs, gain=1.0, inverse=False):
         """Construct an NL class handling non-linearity correction.
 
         Parameters
         ----------
         coeffs : np.ndarray[ncoeff, nx, ny] (float)
             Non-linearity coefficients from reference files.
         """
+        self.inverse = inverse
         self.coeffs = repair_coefficients(coeffs)
-        self.derivcoeffs = derivative(self.coeffs)
-        self.coeffs = self.coeffs[::-1, ...].copy()
-        self.derivcoeffs = self.derivcoeffs[::-1, ...].copy()
-
-    def efficiency(self, counts):
-        """Compute the efficiency of photon counting given a count level.
+        self.gain = gain
```

Comment:
It looks like I used to do self.coeffs = self.coeffs[::-1, ...].copy() around here; was that dropped intentionally?

Reply:
Yes, just to keep the `reverse` keywords defaulting to false. I can swap the logic around, if you prefer.

Reply:
Okay, I see now.  Some part of me felt like it was better to "correct" the mis-ordering on read-in and then have apply the coefficients in the "fast" way (i.e., without reversing again.  I doubt that's actually faster, though.  Okay, sounds good, no action needed.